beep/structure.py

```diff
@@ -1235,39 +1235,27 @@ def determine_structuring_parameters(
         if run_parameter is not None:
             if {"capacity_nominal"}.issubset(run_parameter.columns.tolist()):
                 nominal_capacity = run_parameter["capacity_nominal"].iloc[0]
-            if {"discharge_cutoff_voltage", "charge_cutoff_voltage"}.issubset(
-                run_parameter.columns
-            ):
+            if {"discharge_cutoff_voltage", "charge_cutoff_voltage"}.issubset(run_parameter.columns):
                 v_range = [
                     all_parameters["discharge_cutoff_voltage"].min(),
                     all_parameters["charge_cutoff_voltage"].max(),
                 ]
-            if {
-                "diagnostic_type",
-                "diagnostic_start_cycle",
-                "diagnostic_interval",
-            }.issubset(run_parameter.columns):
+            if {"diagnostic_type", "diagnostic_start_cycle", "diagnostic_interval", }.issubset(run_parameter.columns):
                 if run_parameter["diagnostic_type"].iloc[0] == "HPPC+RPT":
                     hppc_rpt = ["reset", "hppc", "rpt_0.2C", "rpt_1C", "rpt_2C"]
                     hppc_rpt_len = 5
-                    diagnostic_starts_at = [
-                        1,
-                        1
-                        + run_parameter["diagnostic_start_cycle"].iloc[0]
-                        + 1 * hppc_rpt_len,
-                    ]
-                    for i in range(1, 100):
-                        diag_cycle_num = (
-                            i
-                            * (
-                                run_parameter["diagnostic_interval"].iloc[0]
-                                + hppc_rpt_len
-                            )
-                            + 1
-                            + run_parameter["diagnostic_start_cycle"].iloc[0]
-                            + 1 * hppc_rpt_len
-                        )
-                        diagnostic_starts_at.append(diag_cycle_num)
+                    initial_diagnostic_at = [1, 1 + run_parameter["diagnostic_start_cycle"].iloc[0] + 1 * hppc_rpt_len]
+                    assert (np.diff(self.data[self.data.cycle_index == initial_diagnostic_at[0]].index) < 2).all(), \
```

Comment:
@patrickherring-TRI can you explain what's going on here? 
`np.diff(self.data[self.data.cycle_index == initial_diagnostic_at[0]].index) < 2` 

That looks like `np.diff(self.data[ boolean ].index) < 2` which doesn't make any sense.

Reply:
So this is meant to check that the index for the selected cycle is continuous (ie. incrementing by one). The reason to check for this is that a number of PredictionDiagnostics files have the same cycle number at the end of the file where a set of diagnostics was rerun. This should not be happening anymore, and ideally this problem would get caught in the data validator. To solve this issue, it might actually make more sense to reprocess these files and check that they get marked as invalid. Then I can remove this portion of the fix.